pyomo/repn/plugins/gams_writer.py

```diff
@@ -201,46 +312,62 @@ def _print_model( self, model, output_file, solver_capability, var_list,
             var = var_list[i]
             i += 1
             v = symbolMap.getObject(var)
-            try:
+            if not v.is_expression():
                 if v.is_binary():
                     binary.append(var)
                 elif v.is_integer():
                     if v.bounds == (0,1):
                         binary.append(var)
+                    elif v.lb is not None and v.lb >= 0:
+                        posInts.append(var)
                     else:
-                        ints.append(var)
+                        otherInts.append(var)
                 elif v.lb == 0:
                     positive.append(var)
                 else:
                     reals.append(var)
-            except AttributeError:
-                # This is an Expression object
-                body = StringIO()
-                v.to_string(body)
+            else:
+                # Same domain violation check as above
+                try:
+                    value(v.expr)
+                except:
+                    raise RuntimeError(
+                        "While evaluating: %s\n\tProblemWriter_gams"
+                        " encountered an error. Ensure initial"
+                        " variable values\n\tdo not violate"
+                        " any domains (are you using log/log10?)"
```

Comment:
That makes sense.  However, is it the *writer's* responsibility to check this, or is it an error from the solver?  After all, you can (I assume) disable the presolve operations in GAMS.  If you really want to provide additional information to the user, I think I would prefer you to do that as a post-processing operation *if* the solver (i.e., GAMS) returns an error.